### PR TITLE
Basic job board functionality

### PR DESCRIPTION
**Pull request by Kodylow**
I built out the basic client side job board functionality of getting listings for the job board on the left hand side of the Jobs page and displaying a detailed view on the right hand side. I built it out modularly so that eventually if we change the backend from using muse's api to using our own microservice that we can just change the axios route to the new server.

### CODE REVIEW

**Review by juanVmejia — APPROVED** [review]
Everything checked out might consider removing any console logs after using them as eslint will deny merges later on because of them

File: client/src/components/Blog.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import { connect } from 'react-redux'
+
+export const Blog = (props) => {
+  return (
+    <div>
+
```
Comment (cschucode) [inline client/src/components/Blog.jsx:7]: This looks like it should be in a different PR.

File: client/src/components/Jobs/JobDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+const JobDetails = ({ job }) => {
+  return (
+    <div>
+      <h1>{job.name}</h1>
+      <div
+        dangerouslySetInnerHTML={{
```
Comment (cschucode) [inline client/src/components/Jobs/JobDetails.jsx:8]: nice 👍

File: client/src/components/Jobs/JobListing.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import moment from 'moment';
+import React from 'react';
+
+const JobListing = ({ listing, index, setCurrentJob }) => {
+  return (
+    <div
+      className='card'
+      style={{ padding: '20px' }}
```
Comment (cschucode) [inline client/src/components/Jobs/JobListing.jsx:8]: I'm sure this is temporary. I believe we're going with styled components right?

File: client/src/components/Jobs/Jobs.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  const [params, setParams] = useState({});
+  const [page, setPage] = useState(1);
+  const [currentJob, setCurrentJob] = useState(0);
+  const { jobs, loading, error } = fetchJobs(params, page);
+
+  return (
+    <div style={{ display: 'flex' }}>
+      {loading && <h1>Loading...</h1>}
+      {error && <h1>Error</h1>}
+      <div className='listings' style={{ flex: '50%' }}>
```
Comment (cschucode) [inline client/src/components/Jobs/Jobs.jsx:16]: I'd prefer not to use inline styles here and I'm not quite sure this is how flex is used. See https://css-tricks.com/snippets/css/a-guide-to-flexbox/#flex

File: client/src/hooks/fetchJobs.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+const fetchJobs = (params, page) => {
+  const [state, dispatch] = useReducer(reducer, {
+    jobs: [],
+    loading: true,
+    error: false,
+  });
+
+  useEffect(() => {
+    const cancel_Token = axios.CancelToken.source();
```
Comment (cschucode) [inline client/src/hooks/fetchJobs.jsx:35]: What's the purpose of the cancel_Token?

File: jobs/index.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+const express = require('express');
+const cors = require('cors');
+const axios = require('axios');
+
+const app = express();
+app.use(express.json());
+app.use(cors());
+
+app.get('/api/jobs/github', cors(), async (req, res) => {
```
Comment (cschucode) [inline jobs/index.js:9]: @jaylee20  has picked up the ticket for this and will be connecting to the API, storing the data and serving to the front-end.

File: client/src/hooks/fetchJobs.jsx (hunk 5 above)
Comment (Kodylow) [reply client/src/hooks/fetchJobs.jsx:35]: it's for real time search, so while someone's typing it cancels the search calls between letters . This way I just type out the search term and it does the search instead of me having to click "search" which does the request.

**Review by cschucode — APPROVED** [review]

File: client/src/components/Jobs/Jobs.jsx [hunk 7]
```diff
@@ -19,31 +22,52 @@ const Jobs = () => {
...
+      <JobPage>
+        <div>
+          <h3 style={{ textAlign: 'center' }}>
+            {moment().format('dddd, MMMM Do YYYY, h:mm:ss a')}
+          </h3>
+          <Listings>
+            {jobs &&
+              jobs.map((listing, index) => {
+                return (
+                  <JobListing
```
Comment (bry-beep) [inline client/src/components/Jobs/Jobs.jsx]: Good, clear naming

**Review by bry-beep — APPROVED** [review]
Nice work - looks great and it followed the the desired design well. Thanks for the clean starting point for the UI
